solarforecastarbiter/metrics/preprocessing.py

```diff
@@ -0,0 +1,155 @@
+"""
+Provides preprocessing steps to be performed on the timeseries data.
+"""
+
+import pandas as pd
+
+from solarforecastarbiter import datamodel
+from solarforecastarbiter.validation import quality_mapping
+
+
+def apply_validation(data, qfilter, handle_func):
+    """
+    Apply validation steps based on provided filters to the data.
+
+    Parameters
+    ----------
+    qfilter : list of
+        :class:`solarforecastarbiter.datamodel.QualityFlagFilter`
+    data : :class:`pd.DataFrame`
+        Pandas DataFrame of observations and forecasts.
+    handle_func : function
+        Function that handles how `quality_flags` will be used.
+
+    Returns
+    -------
+    pd.DataFrame
+    """
+    validated_data = {}
+
+    # List of flags from filter
+    if not isinstance(qfilter, datamodel.QualityFlagFilter):
+        return TypeError(f"{qfilter} not a QualityFlagFilter")
+    filters = qfilter.quality_flags
+
+    # Apply handling function to quality flags
+    for model, values in data.items():
+
+        # Skip if empty
+        if isinstance(values, pd.DataFrame) and values.empty:
+            validated_data[model] = values.value
+            continue
+
+        # Apply only to Observations
+        if isinstance(model, datamodel.Observation):
+            validation_df = quality_mapping.convert_mask_into_dataframe(
+                values['quality_flag'])
+            validation_df = validation_df[filters]
+            validated_data[model] = handle_func(values.value,
+                                                validation_df.any(axis=1))
+        elif isinstance(model, datamodel.Forecast):
+            validated_data[model] = values
+        else:
+            raise TypeError(f"{model} not an Observation or Forecast")
+
+    return validated_data
+
+
+def resample_and_align(fx_obs, data, tz):
+    """
+    Resample to the forecast and observation using the the larger interval
+    length  and align them to overlap.
+
+    Parameters
+    ----------
+    fx_obs : :class:`solarforecastarbiter.datamodel.ForecastObservation`
+        Pair of forecast and observation.
+    data : dict
+        Dictionary that must include the timeseries of the pair.
+    tz : str
```

Comment:
Glad to hear it. Since reports module imports metrics i like to avoid importing reports in metrics just in case it ever ends up being a circular import.